src/main/java/com/iexec/core/configuration/PublicConfigurationService.java

```diff
@@ -0,0 +1,98 @@
+import org.springframework.stereotype.Service;
+import org.web3j.crypto.Hash;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * This simple service will generate a random session id when the scheduler is started, it will be send to workers when
+ * they ping the scheduler. If they see that the session id has changed, it means that the scheduler has restarted.
+ */
+@Service
+public class PublicConfigurationService {
+    private final ChainConfig chainConfig;
+    private final CredentialsService credentialsService;
+    private final WorkerConfiguration workerConfiguration;
+    private final ResultRepositoryConfiguration resultRepoConfig;
+    private final BlockchainAdapterClientConfig blockchainAdapterClientConfig;
+
+    private PublicConfiguration publicConfiguration = null;
+    /**
+     * {@literal publicConfigurationHash} is a Base64-encoded hash
+     * of all {@link PublicConfiguration} fields,
+     * concatenated with {@literal \n}.
+     */
+    private String publicConfigurationHash = null;
+
+    public PublicConfigurationService(ChainConfig chainConfig,
+                                      CredentialsService credentialsService,
+                                      WorkerConfiguration workerConfiguration,
+                                      ResultRepositoryConfiguration resultRepoConfig,
+                                      BlockchainAdapterClientConfig blockchainAdapterClientConfig) {
+        this.chainConfig = chainConfig;
+        this.credentialsService = credentialsService;
+        this.workerConfiguration = workerConfiguration;
+        this.resultRepoConfig = resultRepoConfig;
+        this.blockchainAdapterClientConfig = blockchainAdapterClientConfig;
+    }
+
+    @PostConstruct
+    void buildPublicConfiguration() {
+        this.publicConfiguration = PublicConfiguration.builder()
+                .workerPoolAddress(chainConfig.getPoolAddress())
+                .blockchainAdapterUrl(blockchainAdapterClientConfig.getUrl())
+                .schedulerPublicAddress(credentialsService.getCredentials().getAddress())
+                .resultRepositoryURL(resultRepoConfig.getResultRepositoryURL())
+                .askForReplicatePeriod(workerConfiguration.getAskForReplicatePeriod())
+                .requiredWorkerVersion(workerConfiguration.getRequiredWorkerVersion())
+                .build();
+
+        // TODO: would be great to put this in Common
+        // (a simple `@ToString` would be sufficient)
+        final String publicConfigurationAsString = String.join("\n",
+                publicConfiguration.getWorkerPoolAddress(),
+                publicConfiguration.getBlockchainAdapterUrl(),
+                publicConfiguration.getSchedulerPublicAddress(),
+                publicConfiguration.getResultRepositoryURL(),
+                publicConfiguration.getAskForReplicatePeriod() + "",
+                publicConfiguration.getRequiredWorkerVersion()
+        );
```

Comment:
Shouldn't we update it right now with `publicConfiguration.toString()`? Or is it too risky?

Reply:
We'll do it in a later PR :)

Reply:
I'm more enclined to do this in a next PR to go fast here.